Form/FormFactory.php

```
@@ -0,0 +1,55 @@
+<?php
+
+namespace EMS\FormBundle\Form;
+
+use EMS\FormBundle\Components\Field\FieldInterface;
+use EMS\FormBundle\FormConfig\FieldConfig;
+use EMS\FormBundle\FormConfig\FormConfig;
+use EMS\FormBundle\FormConfig\FormConfigFactory;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormFactory as SymfonyFormFactory;
+
+class FormFactory
+{
+    /** @var FormConfigFactory */
+    private $formConfigFactory;
+    /** @var SymfonyFormFactory */
+    private $formFactory;
+
+    public function __construct(FormConfigFactory $formConfigFactory, SymfonyFormFactory $formFactory)
+    {
+        $this->formConfigFactory = $formConfigFactory;
+        $this->formFactory = $formFactory;
+    }
+
+    public function create(string $ouuid, string $locale): Form
+    {
+        $config = $this->formConfigFactory->create($ouuid, $locale);
+        $builder = $this->createBuilder($config);
+
+        foreach ($config->getFields() as $fieldConfig) {
+            $field = $this->createField($fieldConfig);
+            $builder->add($fieldConfig->getName(), $field->getFieldClass(), $field->getOptions());
+        }
+
+        return new Form($builder->getForm(), $config);
+    }
+
+    private function createField(FieldConfig $config): FieldInterface
+    {
+        $class = $config->getClass();
+
+        return new $class($config);
```

Same remark as with validations:

} catch (\Throwable $exception) {
            $this->failures[] = sprintf('Field not implemented: %s', $exception->getMessage());
        }

---

The idea is to silently ignore errors for fields that are not available yet. We prefer to render as much as possible of the form instead of giving an error and not rendering a form at all.